src/pydicom/dataelem.py

```diff
@@ -442,6 +448,19 @@ def value(self, val: Any) -> None:
         # * Which str-like VRs can have backslashes in Part 5, Section 6.2
         # * All byte-like VRs
         # * Ambiguous VRs that may be byte-like
+        if isinstance(val, BufferedIOBase):
+            if self.VR not in BUFFERABLE_VRS:
+                raise ValueError(
+                    f"Invalid VR: {self.VR}. Only the following VRs support buffers: {BUFFERABLE_VRS}."
+                )
```

Comment:
The following is a bit more clear
`f"Elements with a VR of '{self.VR}' cannot be used with buffered values, supported VRs are: {BUFFERABLE_VRS}"`